static/js/dist/app/settings.min.js

```diff
@@ -1 +1,247 @@
\ No newline at end of file
+$(document).ready(function () {
+    $('[data-toggle="tooltip"]').tooltip();
+    $("#apiResetForm").submit(function (e) {
+        api.reset()
+            .success(function (response) {
+                user.api_key = response.data
+                successFlash(response.message)
+                $("#api_key").val(user.api_key)
+            })
+            .error(function (data) {
+                errorFlash(data.message)
+            })
+        return false
+    })
+    $("#settingsForm").submit(function (e) {
+        $.post("/settings", $(this).serialize())
+            .done(function (data) {
+                successFlash(data.message)
+            })
+            .fail(function (data) {
+                errorFlash(data.responseJSON.message)
+            })
+        return false
+    })
+    //$("#imapForm").submit(function (e) {
+    $("#savesettings").click(function() {
+        var imapSettings = {}
+        imapSettings.host = $("#imaphost").val()
+        imapSettings.port = $("#imapport").val()
+        imapSettings.username = $("#imapusername").val()
+        imapSettings.password = $("#imappassword").val()
+        imapSettings.enabled = $('#use_imap').prop('checked')
+        imapSettings.tls = $('#use_tls').prop('checked')
+
+        //Advanced settings
+        imapSettings.folder = $("#folder").val()
+        imapSettings.restrict_domain = $("#restrictdomain").val()
+        imapSettings.delete_campaign = $('#deletecampaign').prop('checked')
+
+        
+        //To avoid unmarshalling error in controllers/api/imap.go. It would fail gracefully, but with a generic error.
+        if (imapSettings.host == ""){
+            errorFlash("No IMAP Host specified")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+        if (imapSettings.port == ""){
+            errorFlash("No IMAP Port specified")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+        if (isNaN(imapSettings.port) || imapSettings.port <1 || imapSettings.port > 65535  ){ 
+            errorFlash("Invalid IMAP Port")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+
+        //api.IMAP.post(imapSettings).done(function (data) { // When using this API approach we get an error in the logs "http: TLS handshake error from 127.0.0.1:53858: remote error: tls: unknown certificate"
```

Comment:
I'd be curious to look into this further, since that's definitely not expected.

Reply:
Well, I tested this again and it works now. Not sure why it wasn't before. Updated.